src/index.tsx

```diff
@@ -170,6 +170,43 @@ function hitTest(element: ExcalidrawElement, x: number, y: number): boolean {
   }
 }
 
+function distance(x1: number, x2: number, y1: number, y2: number) {
+  return Math.sqrt(Math.pow(x1 - x2, 2) + Math.pow(y1 - y2, 2));
+}
+
+function resizeTest(
+  element: ExcalidrawElement,
+  x: number,
+  y: number
+): string | false {
+  // For shapes that are composed of lines, we only enable point-selection when the distance
+  // of the click is less than x pixels of any of the lines that the shape is composed of
+  const lineThreshold = 20;
+
+  const x1 = getElementAbsoluteX1(element);
+  const x2 = getElementAbsoluteX2(element);
+  const y1 = getElementAbsoluteY1(element);
+  const y2 = getElementAbsoluteY2(element);
+
+  const hx = x1 + (x2 - x1) / 2;
+  const hy = y1 + (y2 - y1) / 2;
+
+  // (x1, y1) --A-- (x2, y1)
+  //    |D             |B
+  // (x1, y2) --C-- (x2, y2)
+  if (distance(x, hx - 5, y, y1 - 5) < lineThreshold) return "n";
+  if (distance(x, x1 + 5, y, hy - 5) < lineThreshold) return "w";
+  if (distance(x, x2 + 5, y, hy + 5) < lineThreshold) return "e";
+  if (distance(x, hx + 5, y, y2 + 5) < lineThreshold) return "s";
+
+  if (distance(x, x1 - 5, y, y1 - 5) < lineThreshold) return "nw";
+  if (distance(x, x2 + 5, y, y1 - 5) < lineThreshold) return "ne";
+  if (distance(x, x1 + 5, y, y2 + 5) < lineThreshold) return "sw";
+  if (distance(x, x2 + 5, y, y2 + 5) < lineThreshold) return "se";
```

Comment:
I don't think that we should use a line to describe square handles. Instead we should check if the mouse is within a square.

Reply:
Nice!